File: app/src/main/java/com/google/sample/cloudvision/searchpojos/Context.java

package com.google.sample.cloudvision.searchpojos;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Context implements Parcelable
{

    @SerializedName("title")
    @Expose
    private String title;
    public final static Creator<Context> CREATOR = new Creator<Context>() {


        @SuppressWarnings({
            "unchecked"
        })
        public Context createFromParcel(Parcel in) {
            Context instance = new Context();
            instance.title = ((String) in.readValue((String.class.getClassLoader())));
            return instance;
        }

        public Context[] newArray(int size) {
            return (new Context[size]);
        }

    }
    ;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(title);
    }

    public int describeContents() {
        return  0;
    }

}

File: app/src/main/java/com/google/sample/cloudvision/searchpojos/Context.kt
package com.google.sample.cloudvision.searchpojos

import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Context : Parcelable {

    @SerializedName("title")
    @Expose
    var title: String? = null

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(title)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        val CREATOR: Parcelable.Creator<Context> = object : Parcelable.Creator<Context> {


            override fun createFromParcel(`in`: Parcel): Context {
                val instance = Context()
                instance.title = `in`.readValue(String::class.java.classLoader) as String
                return instance
            }

            override fun newArray(size: Int): Array<Context?> {
                return arrayOfNulls(size)
            }

        }
    }

}
